build_tools/embed_data/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+add_executable(generate_cc_embed_data)
```

Comment:
I could see us wrapping the CMake `add_executable` in some sort of `external_cc_binary` util located here: https://github.com/google/iree/tree/master/build_tools/cmake. In general, we want the CMake files to closely match Bazel BUILD files in structure, so we can make mechanical changes (by hand or with a script) to synchronize them.

Maybe I could do the same thing with third_party/dear_imgui : https://github.com/google/iree/blob/0d41620fbe415d5fa3b492a804653c9dea9b8fd5/build_tools/third_party/dear_imgui/CMakeLists.txt#L81-L93 

No action needed in this pull request, just making an observation.